### PR TITLE
Add hyperlane-sovereign framework and hooks

### PR DESCRIPTION
**Pull request by luke-lorenzini**
### Description
Added hooks to accept new Sovereign chain

### Drive-by changes
No

### Related issues
https://github.com/orgs/eigerco/projects/36/views/1?pane=issue&itemId=83625970&issue=eigerco%7Chyperlane-monorepo%7C1

### Backward compatibility
yes

### Testing
NA

### CODE REVIEW

File: rust/main/chains/hyperlane-sovereign/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[package]
+name = "hyperlane-sovereign"
+documentation.workspace = true
+edition.workspace = true
+homepage.workspace = true
+license-file.workspace = true
+publish.workspace = true
+version.workspace = true
```
Comment (asmie) [inline rust/main/chains/hyperlane-sovereign/Cargo.toml]: Is this a must to stick with the whole repository version?
I just mean, could we create new versions of the chain and update its version? If not, how we can distinguish what version of our chain is currently deployed?
I saw that sealevel is keeping its own versioning while the other three chains are deriving it from the workspace 😄

File: rust/main/chains/hyperlane-sovereign/src/mailbox.rs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+use hyperlane_core::{
+    ChainResult, ContractLocator, HyperlaneChain, HyperlaneContract, HyperlaneDomain,
+    HyperlaneMessage, HyperlaneProvider, Mailbox, TxCostEstimate, TxOutcome, H256, U256,
+};
+use std::{
+    fmt::{Debug, Formatter},
+    num::NonZeroU64,
+};
+
+pub struct SovereignMailbox {
```
Comment (asmie) [inline rust/main/chains/hyperlane-sovereign/src/mailbox.rs:15]: Please put a comment for public structure.

File: rust/main/chains/hyperlane-sovereign/src/trait_builder.rs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+use hyperlane_core::config::OperationBatchConfig;
+
+#[derive(Debug, Clone)]
+pub struct ConnectionConf {
```
Comment (asmie) [inline rust/main/chains/hyperlane-sovereign/src/trait_builder.rs:5]: Please put a comment for public structure.

File: rust/main/hyperlane-base/src/settings/chains.rs [hunk 4]
```diff
@@ -113,6 +114,8 @@ pub enum ChainConnectionConf {
     Sealevel(h_sealevel::ConnectionConf),
     /// Cosmos configuration.
     Cosmos(h_cosmos::ConnectionConf),
+    /// Sovereign configuration
```
Comment (asmie) [inline rust/main/hyperlane-base/src/settings/chains.rs]:
```suggestion
    /// Sovereign configuration.
```

File: rust/main/chains/hyperlane-sovereign/src/validator_announce.rs [hunk 5]
```diff
@@ -0,0 +1,46 @@
+use async_trait::async_trait;
+
+use hyperlane_core::{
+    Announcement, ChainResult, HyperlaneChain, HyperlaneContract, HyperlaneDomain,
+    HyperlaneProvider, SignedType, TxOutcome, ValidatorAnnounce, H256, U256,
+};
+/// A reference to a ValidatorAnnounce contract on some Sovereign chain
```
Comment (asmie) [inline rust/main/chains/hyperlane-sovereign/src/validator_announce.rs]:
```suggestion

/// A reference to a ValidatorAnnounce contract on some Sovereign chain.
```

File: rust/main/chains/hyperlane-sovereign/src/trait_builder.rs [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+    /// Operation batching configuration
```
Comment (asmie) [inline rust/main/chains/hyperlane-sovereign/src/trait_builder.rs]:
```suggestion
    /// Operation batching configuration.
```

**Review by asmie — APPROVED** [review]
Very well! 👍 
Only some nitpicks - feel free to resolve them now or with some future PR.

File: rust/main/chains/hyperlane-sovereign/Cargo.toml (hunk 1 above)
Comment (luke-lorenzini) [reply rust/main/chains/hyperlane-sovereign/Cargo.toml]: Good point! Addressed.

**Review by pscott31 — APPROVED** [review]

**Pull request merged**